### ISSUE TITLE
ID Token claims with the are not persisted when using clustering

### ISSUE THREAD

**Issue by RCanine** [Connect2id/Server-Issues#8]
Implementing `op.authz.feedSubjectSessionClaimsIntoIDToken`, we found that claims supplied with:

```
PUT {{c2id}}/authz-sessions/rest/v2/{{asid}}?ajax=true
{
  "sub": "{{sub}}",
  "claims": { ... JSON ... }
}
```

Do not get added to the ID Token. Disabling clustering or resizing the cluster to one instance fixed the issue.

**Comment by c2id-support** [comment]
Thanks for reporting this. We'll try to replicate this, check for any marshalling issues. 

Which Connect2id server version are you using? 

If we've got any other questions regarding replication of this issue we'll get back to you.

**Comment by RCanine** [comment]
This is using version 3.4.

**Comment by c2id-support** [comment]
Hello Ryan,

Could you verify that the `WEB-INF/infinispan.xml` configuration file contains the following lines:

```
<advanced-externalizer class="com.nimbusds.sessionstore.impl.ext.SubjectSessionExternalizerV2"/>
<advanced-externalizer class="com.nimbusds.sessionstore.impl.ext.SubjectSessionExternalizerV1"/>
```

We want to make sure that serialisation of the new "claims" session parameter is enabled, and you're not using an old Infinispan configuration.

**Comment by pakfur** [comment]
I can confirm that those two lines are in WEB-INF/infinispan.xml of the running c2id server

**Comment by c2id-support** [comment]
Thanks!

**Comment by c2id-support** [comment]
Hello,

If your Connect2id cluster consists of v3.4 or v3.3 nodes only (and no nodes running v3.2 or older), remove or uncomment the following XML element from the `WEB-INF/infinispan.xml' config file:

```
<advanced-externalizer class="com.nimbusds.sessionstore.impl.ext.SubjectSessionExternalizerV1"/>
```

This will disable the V1 marshaller for session objects prior to introducing the "claims" field, which Infinispan seems to pick first for unmarshalling session objects (received over the network), despite it being declared after the V2 marshaller and having a smaller internal ID.

The two marshaller versions were put there to ensure a smooth migration from 3.2 and older servers to 3.3+. We'll let you know once we have a fix for that.

**Comment by c2id-support** [comment]
Hello,

We patched up user session marshalling in 3.4, which is now no longer dependent on Infinispan's algorithm for prioritising marshallers for a given class (which we suspect is non-deterministic). The new "claims" field in user session objects should now replicate without problems across the cluster.

https://connect2id.com/assets/products/server/download/3.4.3/Connect2id-server.zip
https://connect2id.com/assets/products/server/download/3.4.3/c2id.war

Your Connect2id support team
